remoteworkitem/tracker_repository.go

```diff
@@ -101,107 +92,67 @@ func (r *GormTrackerRepository) CheckExists(ctx context.Context, id string) erro
-			URL:  tracker.URL,
-			Type: tracker.Type}
-		result[i] = &t
-	}
-	return result, nil
+func (r *GormTrackerRepository) List(ctx context.Context) ([]Tracker, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "tracker", "query"}, time.Now())
+	var objs []Tracker
+	err := r.db.Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, err
+	}
+	return objs, nil
 }
 
 // Save updates the given tracker in storage.
 // returns NotFoundError, ConversionError or InternalError
-func (r *GormTrackerRepository) Save(ctx context.Context, t app.Tracker) (*app.Tracker, error) {
+func (r *GormTrackerRepository) Save(ctx context.Context, t *Tracker) (*Tracker, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "tracker", "save"}, time.Now())
 	res := Tracker{}
-	id, err := strconv.ParseUint(t.ID, 10, 64)
-	if err != nil || id == 0 {
-		return nil, NotFoundError{entity: "tracker", ID: t.ID}
-	}
-
-	log.Info(ctx, map[string]interface{}{
-		"tracker_id": id,
-	}, "Looking for a tracker repository with id ", id)
-
-	tx := r.db.First(&res, id)
+	tx := r.db.Where("id = ?", t.ID).Find(&res)
 	if tx.RecordNotFound() {
 		log.Error(ctx, map[string]interface{}{
-			"tracker_id": id,
+			"tracker_id": t.ID,
 		}, "tracker repository not found")
-
-		return nil, NotFoundError{entity: "tracker", ID: t.ID}
+		return nil, errors.NewNotFoundError("tracker", t.ID.String())
 	}
 	_, present := RemoteWorkItemImplRegistry[t.Type]
 	// Ensure we support this remote tracker.
 	if present != true {
-		return nil, BadParameterError{parameter: "type", value: t.Type}
+		return nil, errors.NewBadParameterError("type", t.Type)
 	}
 
-	newT := Tracker{
-		ID:   id,
-		URL:  t.URL,
-		Type: t.Type}
-
-	if err := tx.Save(&newT).Error; err != nil {
+	if err := tx.Save(&t).Error; err != nil {
 		log.Error(ctx, map[string]interface{}{
-			"tracker_id": newT.ID,
+			"tracker_id": t.ID,
 			"err":        err,
 		}, "unable to save tracker repository")
-		return nil, InternalError{simpleError{err.Error()}}
+		return nil, errors.NewInternalError(ctx, err)
 	}
-
-	log.Info(ctx, map[string]interface{}{
-		"tracker": newT.ID,
-	}, "Tracker repository successfully updated")
-
-	t2 := app.Tracker{
-		ID:   strconv.FormatUint(id, 10),
-		URL:  t.URL,
-		Type: t.Type}
-
-	return &t2, nil
+	return t, nil
 }
 
 // Delete deletes the tracker with the given id
 // returns NotFoundError or InternalError
-func (r *GormTrackerRepository) Delete(ctx context.Context, ID string) error {
-	var t = Tracker{}
-	id, err := strconv.ParseUint(ID, 10, 64)
-	if err != nil || id == 0 {
-		// treat as not found: clients don't know it must be a number
-		return NotFoundError{entity: "tracker", ID: ID}
+func (r *GormTrackerRepository) Delete(ctx context.Context, ID uuid.UUID) error {
+	defer goa.MeasureSince([]string{"goa", "db", "tracker", "delete"}, time.Now())
+	if ID == uuid.Nil {
+		log.Error(ctx, map[string]interface{}{
+			"tracker_id": ID.String(),
```

Comment:
can you please log the `error` as well? `"err": err`
at other locations as well.

Reply:
fixed it here ->
71b44b5